tests/system/test_bucket.py

```diff
@@ -87,9 +94,21 @@ def test_bucket_lifecycle_rules(storage_client, buckets_to_delete):
     assert bucket.name == bucket_name
     assert list(bucket.lifecycle_rules) == expected_rules
 
+    # Test modifying lifecycle rules
+    expected_rules[0] = LifecycleRuleDelete(age=30)
+    rules = list(bucket.lifecycle_rules)
+    rules[0]["condition"] = {"age": 30}
+    bucket.lifecycle_rules = rules
+    bucket.patch()
+
+    bucket.reload()
```

Comment:
Curious do we need to reload the bucket here, same on L111? The patch call should return the bucket metadata? Otherwise, LGTM, thanks!

Reply:
No, you're right, we should be able to skip that. I'll remove them and confirm it still works.